pagopa/io-app
# chore: [IOBP-629] Add onboarding payment method when not found any

## Short description
This PR implements the possibility to add a new payment method during the checkout, being able to resume the payment flow once the method has been successfully added.

**N.B:** ⚠ This behavior has been implemented for a transitional phase with the current PM, as soon as the backend is fully migrated to NPG, this functionality will become deprecated and thus must be removed (reported in the following ticket: https://pagopa.atlassian.net/browse/IOBP-632)

## List of changes proposed in this pull request
- Added a new action to save the `rptId` of the payment in order to resume the transaction later after the onboarding is completed successfully.
- Added a check into [PaymentsOnboardingFeedbackScreen.tsx](https://github.com/pagopa/io-app/compare/master...IOBP-629-add-payment-method-when-not-found-any#diff-9856420745ec55e65f6a52dd2ffc690d46822ac462925108abc93f6a928c4874), if is present a `rptIdToResume` and the outcome is success, It starts the payment flow giving as input the `rptIdToResume`.
- Added an effect into [WalletPaymentPickMethodScreen.tsx](https://github.com/pagopa/io-app/compare/master...IOBP-629-add-payment-method-when-not-found-any#diff-08aa31e861c2cc684eaec44fc714e461c252a9bb06f9d7971429bbdf756af92d) that if the PM doesn't give any payment method onboarded and neither a list of all onboardable payment methods, so it navigates to an outcome screen that shows the CTA to start the onboarding.

## How to test
- Open the new `Payments` section with the feature flag enabled;
- Remove all the payment methods already onboarded;
- Start a new payment flow setting into the `io-dev-api-server` a list of empty payment-methods
- You should be able to onboard a new card and then resume the payment flow

## Preview

https://github.com/pagopa/io-app/assets/34343582/4c524682-61ac-497a-8804-fa7aca3110db
